qiskit/algorithms/time_evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,265 @@
+                rounded up to 0 for quantities that are expected to be
+                non-negative.
+        """
+        super().__init__()
+        self.ansatz = ansatz
+        self.initial_parameters = initial_parameters
+        self.variational_principle = variational_principle
+        self.estimator = estimator
+        self.num_timesteps = num_timesteps
+        self.lse_solver = lse_solver
+        # OdeFunction abstraction kept for potential extensions - unclear at the moment;
+        # currently hidden from the user
+        self._ode_function_factory = OdeFunctionFactory(lse_solver=lse_solver)
+        self.ode_solver = ode_solver
+        self.imag_part_tol = imag_part_tol
+        self.num_instability_tol = num_instability_tol
+
+    def evolve(self, evolution_problem: TimeEvolutionProblem) -> VarQTEResult:
+        """Apply Variational Quantum Time Evolution w.r.t. the given operator.
+
+        Args:
+            evolution_problem: Instance defining an evolution problem.
+        Returns:
+            Result of the evolution which includes a quantum circuit with bound parameters as an
+            evolved state and, if provided, observables evaluated on the evolved state.
+
+        Raises:
+            ValueError: If ``initial_state`` is included in the ``evolution_problem``.
+        """
+        self._validate_aux_ops(evolution_problem)
+
+        if evolution_problem.initial_state is not None:
+            raise ValueError("initial_state provided but not applicable to VarQTE.")
+
+        init_state_param_dict = self._create_init_state_param_dict(
+            self.initial_parameters, self.ansatz.parameters
+        )
+
+        # unwrap PauliSumOp (in the future this will be deprecated)
+        if isinstance(evolution_problem.hamiltonian, PauliSumOp):
+            hamiltonian = (
+                evolution_problem.hamiltonian.primitive * evolution_problem.hamiltonian.coeff
+            )
+        else:
+            hamiltonian = evolution_problem.hamiltonian
+
+        evolved_state, param_values, time_points = self._evolve(
+            init_state_param_dict,
+            hamiltonian,
+            evolution_problem.time,
+            evolution_problem.t_param,
+        )
+
+        observables = []
+        if evolution_problem.aux_operators is not None:
+            for values in param_values:
+                # cannot batch evaluation because estimate_observables
+                # only accepts single circuits
+                evol_state = self.ansatz.assign_parameters(
+                    dict(zip(init_state_param_dict.keys(), values))
+                )
+                observable = estimate_observables(
+                    self.estimator,
+                    evol_state,
+                    evolution_problem.aux_operators,
+                )
+                observables.append(observable)
+
+        # TODO: deprecate returning evaluated_aux_ops.
+        #  As these are the observables for the last time step.
```

Comment:
Is this a future TODO?

Reply:
Yes. We decided to keep returning the last evaluated observable to keep consistency with the already released trotter evolution result, but it is a bit redundant if we are already returning the list of observables. I think this is fine for the time being, but left the TODO as a reminder to revisit this topic in the future.